### PR TITLE
Add Documentation Guidelines

### PR DESCRIPTION
**Pull request by samhita-alla**
Signed-off-by: Samhita Alla <aallasamhita@gmail.com>

# TL;DR
Updated Flytekit contribution guide.

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

File: docs/source/contributing.rst [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
-Flytekit Contributor Guide
-##########################
+###########################
+Flytekit Contribution Guide
+###########################
 
 First off, thank you for thinking about contributing! Below you'll find instructions that will hopefully guide you through how to fix, improve, and extend Flytekit.
 
-Please also take some time to read through the design guides, which describe the various parts of Flytekit and should make contributing easier.
+Please also take some time to read through the `design guides <https://docs.flyte.org/projects/flytekit/en/latest/design/index.html>`__, which describe the various parts of Flytekit and should make contributing easier.
```
Comment (wild-endeavor) [inline docs/source/contributing.rst:9]: can we use the sphinx link for this?
```
:std:ref:`design guides <design>`
```

i think it's that.

File: docs/source/contributing.rst [hunk 2]
```diff
@@ -1,16 +1,19 @@
 .. _contributing:
 
-##########################
-Flytekit Contributor Guide
-##########################
+###########################
+Flytekit Contribution Guide
```
[the remark this answers is not in the document]
Comment (samhita-alla) [reply docs/source/contributing.rst:4]: Yeah, it's "contribution guide" everywhere.

File: docs/source/contributing.rst (hunk 1 above)
Comment (samhita-alla) [reply docs/source/contributing.rst:9]: Done!

**Comment by wild-endeavor** [comment]
lint is failing?

**Comment by wild-endeavor** [comment]
and just something in the PR description?

**Comment by samhita-alla** [comment]
@wild-endeavor,
> lint is failing?

Fixed.
> and just something in the PR description?

Added.

**Review by wild-endeavor — APPROVED** [review]

**Pull request merged**